Utils/Dataflow/pyDKB/common/misc.py

```diff
@@ -46,3 +47,25 @@ def log(message, level=logLevel.INFO, *args):
             out_message += "\n(==) %s" % l
         out_message += "\n"
         sys.stderr.write(out_message)
+
+
+def try_to_import(modname, attrname=None):
+    """ Try to import specified module or attribute from a module.
+
+    If module/attribute can not be imported, catch the exception and output log
+    message.
+    """
+    result = False
+    try:
+        result = importlib.import_module(modname)
+        if attrname:
+            result = getattr(result, attrname)
+    except ImportError, err:
+        log("Failed to import '%s'.\nDetails: %s" % (modname, err),
+            logLevel.ERROR)
+    except AttributeError:
+        if attrname:
+            log("Failed to import '%s' from '%s'" % (attrname, modname))
```

Comment:
1. In this case module will be returned as `result`. What's the point of returning the whole module when an attribute is expected? Maybe result should stay `False`?
2. Shouldn't this message be logged as ERROR, and not INFO?
3. Shouldn't the error message be displayed?

Reply:
@Evildoor,

1-2: fair point, fixed in 6360f5d.

3: a bit unclear, but still a fair point too -- if you meant that in case of module failed to be imported due to the `AttributeError` we kind of ignore this error, silently returning `False`.
Fixed this in 5891f41 (and, coincidentally, made code much simplier). Also made return value less confusing: in case of error function will return its own specific value `NOT_IMPORTED`, defined in same module, instead of `False` (see 10ac41a).

---

*Please take a look and mark this conversation as solved if everything is OK.*